### PR TITLE
[PAY-866] Add new remix settings - Part 1

### PR DESCRIPTION
**Pull request by sddioulde**
### Description

Add new remix settings and handle relationship with premium track availability settings.

Will merge to main after https://github.com/AudiusProject/audius-client/pull/2752 is merged into this one.

### Dragons

track upload flow

### How Has This Been Tested?

local dapp v stage
test remix settings work as currently expected
set track as premium, check remix settings state
set track as remixabled, check track availability state

### How will this change be monitored?

n/a

### Feature Flags ###

exisiting premium content feature flags

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/sd-pc-settings

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/sd-pc-settings

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/sd-pc-settings

**Comment by sddioulde** [comment]
lint is fixed when https://github.com/AudiusProject/audius-client/pull/2752 is merged into here

File: packages/web/src/components/modal-radio/ModalRadioGroup.tsx [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { ReactNode } from 'react'
```
Comment (rickyrombo) [inline packages/web/src/components/modal-radio/ModalRadioGroup.tsx:1]: [packages/stems/src/components/RadioButtonGroup/RadioButtonGroup.tsx](https://github.com/AudiusProject/audius-client/blob/937b1886cdb932649571cb13db22640b47d07e12/packages/stems/src/components/RadioButtonGroup/RadioButtonGroup.tsx)?

File: packages/web/src/components/track-availability-modal/CollectibleGatedAvailability.tsx [hunk 2]
```diff
@@ -85,87 +93,84 @@ export const CollectibleGatedAvailability = ({
...
+        <div className={styles.noCollectibles}>
+          <IconQuestionCircle className={styles.noCollectibleIcon} />
+          <div className={styles.noCollectibleText}>
+            {messages.noCollectibles}
+          </div>
         </div>
+      )}
+      <div
+        className={styles.learnMore}
+        onClick={() => window.open(LEARN_MORE_URL, '_blank')}
```
Comment (rickyrombo) [inline packages/web/src/components/track-availability-modal/CollectibleGatedAvailability.tsx:119]: Use a real `<a>` with href?

File: packages/web/src/components/track/PremiumTrackSection.tsx [hunk 3]
```diff
@@ -84,20 +84,20 @@ const LockedPremiumTrackSection = ({
   goToCollection
 }: PremiumTrackAccessSectionProps) => {
   const dispatch = useDispatch()
-  const sendStatus = useSelector(getSendStatus)
-  const previousSendStatus = usePrevious(sendStatus)
+  const tipSendStatus = useSelector(getSendStatus)
+  const previousSendStatus = usePrevious(tipSendStatus)
```
Comment (rickyrombo) [inline packages/web/src/components/track/PremiumTrackSection.tsx:88]: nit: maybe rename `previousSendStatus` => `previousTipSendStatus`?

**Review by rickyrombo — APPROVED** [review]

File: packages/web/src/components/track-availability-modal/TrackAvailabilityModal.module.css [hunk 4]
```diff
@@ -142,26 +103,24 @@
...
-  width: 304px;
+.hiddenSectionColumn:first-child {
   margin-right: 12px;
 }
 
-.availabilityRowSelection > div:not(:last-child),
-.availabilityRowSelection > div > div:not(:last-child) {
-  margin-bottom: 12px;
+.hiddenSectionColumn:last-child {
+  margin-left: 12px;
```
Comment (dharit-tan) [inline packages/web/src/components/track-availability-modal/TrackAvailabilityModal.module.css:123]: nit: using --var(--unit-3) etc.

File: packages/web/src/components/modal-radio/ModalRadioGroup.tsx (hunk 1 above)
Comment (sddioulde) [reply packages/web/src/components/modal-radio/ModalRadioGroup.tsx:1]: this is different.
the radio button here is new in web. no bandwidth to move to stems atm but that's the goal, and also have a better name for it, but for now it's only in some modals.

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/sd-pc-settings

File: packages/web/src/components/modal-radio/ModalRadioGroup.tsx (hunk 1 above)
Comment (rickyrombo) [reply packages/web/src/components/modal-radio/ModalRadioGroup.tsx:1]: Are they different only by style? The `RadioButtonGroup` component is hopefully versatile enough if not the `RadioButton` itself

**Pull request merged**